pkg/volumes/openstack/metrics.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openstack
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type OpenstackMetrics struct {
+	Duration *prometheus.HistogramVec
+	Total    *prometheus.CounterVec
+	Errors   *prometheus.CounterVec
+}
+
+// MetricContext indicates the context for OpenStack metrics.
+type MetricContext struct {
+	Start      time.Time
+	Attributes []string
+	Metrics    *OpenstackMetrics
+}
+
+// NewMetricContext creates a new MetricContext.
+func NewMetricContext(resource string, request string) *MetricContext {
+	return &MetricContext{
+		Start:      time.Now(),
```

Comment:
Although the context is always created directly before using `Observe` I'd recommend moving the Start initialisation into `Observe` to prevent accidental reuse of that context and getting the wrong durations out of it.

Reply:
its not same thing? The idea here is following:

1) newmetriccontext - take time stamp
2) execute some function
3) observe and calculate duration

if we move time.now (start) to observe we cannot calculate duration anymore

Reply:
OK, I just saw how it is used.